app/controllers/journeys/nics/Class4NICsController.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.journeys.nics
+
+import controllers.actions._
+import controllers.journeys.fillForm
+import forms.standard.BooleanFormProvider
+import models.Mode
+import models.common.{BusinessId, TaxYear}
+import pages.nics.Class4NICsPage
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import services.SelfEmploymentService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import views.html.journeys.nics.Class4NICsView
+
+import javax.inject.{Inject, Singleton}
+
+@Singleton
+class Class4NICsController @Inject() (override val messagesApi: MessagesApi,
+                                      val controllerComponents: MessagesControllerComponents,
+                                      identify: IdentifierAction,
+                                      getData: DataRetrievalAction,
+                                      requireData: DataRequiredAction,
+                                      formProvider: BooleanFormProvider,
+                                      service: SelfEmploymentService,
+                                      view: Class4NICsView)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  private val page = Class4NICsPage
+
+  def onPageLoad(taxYear: TaxYear, businessId: BusinessId, mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
+    implicit request =>
+      val form = fillForm(page, businessId, formProvider(page, request.userType))
+      Ok(view(form, taxYear, businessId, request.userType, mode))
+  }
+
+  def onSubmit(taxYear: TaxYear, businessId: BusinessId, mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      def handleError(formWithErrors: Form[_]): Result = BadRequest(view(formWithErrors, taxYear, businessId, request.userType, mode))
+
+      service.defaultHandleForm(formProvider(page, request.userType), page, businessId, taxYear, mode, handleError)
+  }
```

Comment:
Have a look at GoodsAndServicesForYourOwnUseController submit method, which handles the boolean submission for a gateway question (will then clear data for skipped pages)